test/clustertest/testplugin/TestPlugin.cpp

```diff
@@ -21,6 +22,52 @@ BedrockPlugin_TestPlugin::~BedrockPlugin_TestPlugin()
 {
 }
 
+bool FileAppend(const string& path, const string& buffer) {
+    // Try to open the file for appending.
+    FILE* fp = fopen(path.c_str(), "a");
+    if (!fp) {
+        return false;
+    }
+
+    // Lock, nobody else can write.
+    flock(fileno(fp), LOCK_EX);
+
+    // Write.
+    size_t numWritten = fwrite(buffer.c_str(), 1, buffer.size(), fp);
```

Comment:
should we wrap this in a try/catch, then use the catch to call `flock(flock(fileno(fp), LOCK_UN));` and `fclose(fp)`. Otherwise it's conceivable we leak locks and fps if we crash at this line. same thing for L56-64 in the next function. Not sure if that's too defensive or not 🤔

Reply:
Nothing in here throws, it's all C stuff. If it crashes it's gonna segfault or abort and we won't catch it anyway.